src/ext/tags.js

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2016 Resonance Labs, Inc
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+// Here we define standard names for tags that can be added to spans by the
+// instrumentation code. The actual tracing systems are not required to
+// retain these as tags in the stored spans if they have other means of
+// representing the same data. For example, the SPAN_KIND:'server' can be
+// inferred from a Zipkin span by the presence of ss/sr annotations.
+// For more information visit http://opentracing.io/data-semantics/
+
+module.exports = {
+
+    /** ------------------------------------------------------------------------
+    * SPAN_KIND hints at relationship between spans, e.g. client/server
+    * --------------------------------------------------------------------------*/
+    SPAN_KIND : 'span.kind',
+
+    // Marks a span representing the client-side of an RPC or other remote call
+    SPAN_KIND_RPC_CLIENT : 'client',
+
+    // Marks a span representing the server-side of an RPC or other remote call
+    SPAN_KIND_RPC_SERVER : 'server',
+
+    /** ------------------------------------------------------------------------
+    * ERROR (boolean) indicates whether a Span ended in an error state.
+    * --------------------------------------------------------------------------*/
+    ERROR : 'error',
+
+    /** ------------------------------------------------------------------------
+    * COMPONENT (string) ia s low-cardinality identifier of the module, library,
+    * or package that is generating a span.
+    * --------------------------------------------------------------------------*/
+    COMPONENT : 'component',
+
+    /** ------------------------------------------------------------------------
+    * SAMPLING_PRIORITY (uint16) determines the priority of sampling this Span.
```

Comment:
these integer widths don't mean anything in .js... let's just call them all "number"s